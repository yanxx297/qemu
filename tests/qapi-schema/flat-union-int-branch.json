# we require flat union branches to be a struct
{ 'enum': 'TestEnum',
  'data': [ 'value1', 'value2' ] }
{ 'type': 'Base',
  'data': { 'enum1': 'TestEnum' } }
{ 'type': 'TestTypeB',
  'data': { 'integer': 'int' } }
{ 'union': 'TestUnion',
  'base': 'Base',
  'discriminator': 'enum1',
  'data': { 'value1': 'int',
            'value2': 'TestTypeB' } }
