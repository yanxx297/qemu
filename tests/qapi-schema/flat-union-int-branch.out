[OrderedDict([('enum', 'TestEnum'), ('data', ['value1', 'value2'])]),
 OrderedDict([('type', 'Base'), ('data', OrderedDict([('enum1', 'TestEnum')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))]),
 OrderedDict([('union', 'TestUnion'), ('base', 'Base'), ('discriminator', 'enum1'), ('data', OrderedDict([('value1', 'int'), ('value2', 'TestTypeB')]))])]
[{'enum_name': 'TestEnum', 'enum_values': ['value1', 'value2']}]
[OrderedDict([('type', 'Base'), ('data', OrderedDict([('enum1', 'TestEnum')]))]),
 OrderedDict([('type', 'TestTypeB'), ('data', OrderedDict([('integer', 'int')]))])]
